packages/frontend/src/features/metricsCatalog/hooks/useDateRangePicker.tsx

```diff
@@ -367,26 +370,31 @@ export const useDateRangePicker = ({
                             }
                         },
                         numberOfColumns: 2,
-                        firstDayOfWeek: 0,
+                        firstDayOfWeek: 1,
                         hideOutsideDates: true,
                         // Highlight the week of the selected date
                         getDayProps: (date) => {
+                            const today = dayjs().endOf('day');
+                            const isInFuture = dayjs(date).isAfter(today);
+
                             const isSelected = Boolean(
                                 tempDateRange[0] &&
                                     tempDateRange[1] &&
                                     date >=
                                         dayjs(tempDateRange[0])
-                                            .startOf('week')
+                                            .startOf('isoWeek')
                                             .toDate() &&
                                     date <=
                                         dayjs(tempDateRange[1])
-                                            .endOf('week')
-                                            .toDate(),
+                                            .endOf('isoWeek')
+                                            .toDate() &&
+                                    !isInFuture, // Don't highlight future dates
                             );
+
                             return {
                                 inRange: isSelected,
-                                firstInRange: date.getDay() === 0 && isSelected, // Highlight the first day of the week
-                                lastInRange: date.getDay() === 6 && isSelected, // Highlight the last day of the week
+                                firstInRange: date.getDay() === 1 && isSelected,
```

Comment:
starts on monday. we should listen to warehouse settings - sunday or monday - but that can be done on a separate PR / in the future